src/coordinator/api/v1/handler/prometheus/remote/write.go

```diff
@@ -44,18 +48,31 @@ const (
 	PromWriteHTTPMethod = http.MethodPost
 )
 
+var (
+	errNoStorageOrDownsampler = errors.New("no storage or downsampler set, requires at least one or both")
+)
+
 // PromWriteHandler represents a handler for prometheus write endpoint.
 type PromWriteHandler struct {
 	store            storage.Storage
+	downsampler      downsample.Downsampler
 	promWriteMetrics promWriteMetrics
 }
 
 // NewPromWriteHandler returns a new instance of handler.
-func NewPromWriteHandler(store storage.Storage, scope tally.Scope) http.Handler {
+func NewPromWriteHandler(
+	store storage.Storage,
+	downsampler downsample.Downsampler,
+	scope tally.Scope,
+) (http.Handler, error) {
+	if store == nil && downsampler == nil {
```

Comment:
curious about this condition. Traditionally, we don't check if arguments passed to a struct are null except if they are coming from a config.

Reply:
I prefer to be defensive.